src/bench/wallet_available_coins.cpp

```diff
@@ -0,0 +1,128 @@
+        // add it to the chain
+        context.chainman->ActiveChain().SetTip(pindex);
+    }
+
+    // notify wallet
+    wallet.blockConnected(block, tip.tip_height);
+}
+
+static void WalletAvailableCoins(benchmark::Bench& bench, const std::vector<OutputType>& output_type)
+{
+    const auto test_setup = MakeNoLogFileContext<const TestingSetup>();
+
+    CWallet wallet{test_setup->m_node.chain.get(), "", gArgs, CreateMockWalletDatabase()};
+    {
+        LOCK(wallet.cs_wallet);
+        if (std::any_of(output_type.begin(), output_type.end(), [](OutputType type){return type > OutputType::LEGACY;})) {
+            wallet.SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+            wallet.SetupDescriptorScriptPubKeyMans();
+        }
+        if (std::any_of(output_type.begin(), output_type.end(), [](OutputType type){return type == OutputType::LEGACY;})) {
+            wallet.SetupLegacyScriptPubKeyMan();
+        }
+        if (wallet.LoadWallet() != DBErrors::LOAD_OK) assert(false);
+    }
+
+    // Generate destinations
+    std::vector<CScript> dest_wallet;
+    for (auto type : output_type) {
+        dest_wallet.emplace_back(GetScriptForDestination(getNewDestination(wallet, type)));
+    }
+
+    // Generate chain; each coinbase will have two outputs to fill-up the wallet
+    const auto& params = Params();
+    unsigned int chain_size = 1000;
+    for (unsigned int i = 0; i < chain_size / dest_wallet.size(); ++i) {
+        for (auto dest : dest_wallet) {
+            generateFakeBlock(params, test_setup->m_node, wallet, dest);
+        }
+    }
+
+    // Check available balance
+    auto bal = wallet::GetAvailableBalance(wallet); // Cache
+    assert(bal == 50 * COIN * (chain_size - COINBASE_MATURITY));
+
+    bench.epochIterations(2).run([&] {
+        LOCK(wallet.cs_wallet);
+        std::vector<wallet::COutput> vCoins;
+        wallet::AvailableCoins(wallet, vCoins);
+
+        assert(vCoins.size() == (chain_size - COINBASE_MATURITY) * 2);
+    });
+}
+
+static void WalletAvailableCoinsOnlyBech32M(benchmark::Bench& bench) { WalletAvailableCoins(bench, {OutputType::BECH32M}); }
+static void WalletAvailableCoinsOnlyBech32(benchmark::Bench& bench) { WalletAvailableCoins(bench, {OutputType::BECH32}); }
+static void WalletAvailableCoinsOnlyP2SH_SEGWIT(benchmark::Bench& bench) { WalletAvailableCoins(bench, {OutputType::P2SH_SEGWIT}); }
+static void WalletAvailableCoinsOnlyLegacy(benchmark::Bench& bench) { WalletAvailableCoins(bench, {OutputType::LEGACY}); }
+static void WalletAvailableCoinsMulti(benchmark::Bench& bench) { WalletAvailableCoins(bench, {OutputType::LEGACY, OutputType::BECH32M, OutputType::BECH32, OutputType::P2SH_SEGWIT}); }
+
+BENCHMARK(WalletAvailableCoinsOnlyBech32M);
+BENCHMARK(WalletAvailableCoinsOnlyBech32);
+BENCHMARK(WalletAvailableCoinsOnlyP2SH_SEGWIT);
+BENCHMARK(WalletAvailableCoinsOnlyLegacy);
+BENCHMARK(WalletAvailableCoinsMulti);
```

Comment:
In b791c2ae2eb02ff53762ac5ed2b355e5248ac041 "bench: add benchmark for wallet 'AvailableCoins' function."

I don't really think it is useful to have `AvailableCoins` run for different `OutputTypes`. There really isn't any difference for how `AvailableCoins` behaves depending on the `OutputType`.

Reply:
Hmm, wouldn't we be having the same benchmark times for all the cases if that would be the case? (we are adding the same amount of txs and outputs to each of them).

I mean, your #24699 improves exactly this. The time difference there is for the solving provider and `IsMine` related calls.

Reply:
Yes, it should be the same benchmark times for all cases, so it doesn't make sense to run it separately for each `OutputType`.